### PR TITLE
ENH Updates docs and tests for classification modules that already support no batch dims

### PR DESCRIPTION
**Pull request by thomasjpfan**
Towards #60585

### CODE REVIEW

File: torch/testing/_internal/common_nn.py [hunk 1]
```diff
@@ -4891,6 +4891,50 @@ def padding3d_circular(input, pad):
...
+    ('MultiLabelMarginLoss', lambda: torch.randn(4), lambda: torch.tensor([3, 0, -1, 1])),
+    ('SoftMarginLoss', lambda: torch.randn(9), lambda: torch.tensor([-1, 1, 1] * 3)),
+]
+classification_criterion_no_batch_extra_info: Dict[str, dict] = {
+    'MultiLabelMarginLoss': {'check_gradgrad': False},
+}
+reductions = ['none', 'mean', 'sum']
+for (name, input_fn, target_fn), reduction in product(classification_criterion_no_batch,
+                                                      reductions):
+    classificaiton_test_info = dict(
```
Comment (jbschlosser) [inline torch/testing/_internal/common_nn.py:4927]: spelling nit: classificaiton_test_info -> classification_test_info

File: torch/nn/modules/loss.py [hunk 2]
```diff
@@ -445,9 +445,9 @@ class KLDivLoss(_Loss):
         In the next major release, ``'mean'`` will be changed to be the same as ``'batchmean'``.
 
     Shape:
-        - Input: :math:`(N, *)` where :math:`*` means, any number of additional
-          dimensions
-        - Target: :math:`(N, *)`, same shape as the input
+        - Input: :math:`(*)`, where :math:`*` means, any number of additional
```
Comment (jbschlosser) [inline torch/nn/modules/loss.py:448]: grammar nit: no comma needed after "means" throughout

File: torch/nn/modules/loss.py [hunk 3]
```diff
@@ -688,8 +688,8 @@ class BCEWithLogitsLoss(_Loss):
                 Must be a vector with length equal to the number of classes.
 
     Shape:
-        - Input: :math:`(N, *)` where :math:`*` means, any number of additional dimensions
-        - Target: :math:`(N, *)`, same shape as the input
+        - Input: :math:`(*)`. where :math:`*` means, any number of additional dimensions.
```
Comment (jbschlosser) [inline torch/nn/modules/loss.py:691]: nit: comma instead of period before "where"

File: torch/testing/_internal/common_nn.py [hunk 4]
```diff
@@ -4891,6 +4891,50 @@ def padding3d_circular(input, pad):
...
+
+    if reduction == 'none':
+        return output.squeeze(0)
+    # reduction is 'sum' or 'mean' which results in a scalar
+    return output
+
+
+# Check that classification criterion work with no batch dimensions
+# List of tuples of (name, input_fn, target_fn)
+classification_criterion_no_batch = [
```
Comment (jbschlosser) [inline torch/testing/_internal/common_nn.py:4913]: This works, but it'll be much easier to define tests over sets of modules like this once the `@modules` decorator  / `ModuleInfo` are in :)

**Review by jbschlosser — COMMENTED** [review]
Thanks for the updates! Just some typo comments below. Tests look great!

File: torch/testing/_internal/common_nn.py [hunk 5]
```diff
@@ -4986,6 +4986,49 @@ def padding3d_circular(input, pad):
...
+}
+reductions = ['none', 'mean', 'sum']
+for (name, input_fn, target_fn), reduction in product(classification_criterion_no_batch,
+                                                      reductions):
+    classification_test_info = dict(
+        fullname="{}_no_batch_dim_{}".format(name, reduction),
+        constructor=lambda *args: getattr(nn, name)(reduction=reduction),
+        input_fn=input_fn,
+        target_fn=target_fn,
+        test_cpp_api_parity=False,
```
Comment (jbschlosser) [inline torch/testing/_internal/common_nn.py:5026]: Should `single_batch_reference_criterion_fn` be referenced here?

File: torch/testing/_internal/common_nn.py [hunk 6]
```diff
@@ -4986,6 +4986,49 @@ def padding3d_circular(input, pad):
...
+    )
+    extra_info = classification_criterion_no_batch_extra_info.get(name, {})
+    classification_test_info.update(extra_info)
+    criterion_tests.append(classification_test_info)
+
 
 def single_batch_reference_criterion_fn(*args):
```
Comment (jbschlosser) [inline torch/testing/_internal/common_nn.py:5033]: Removing this re-definition should fix the lint check

File: torch/nn/modules/loss.py [hunk 7]
```diff
@@ -684,9 +683,9 @@ class BCEWithLogitsLoss(_Loss):
                 Must be a vector with length equal to the number of classes.
 
     Shape:
-        - Input: :math:`(N, *)` where :math:`*` means, any number of additional dimensions
-        - Target: :math:`(N, *)`, same shape as the input
-        - Output: scalar. If :attr:`reduction` is ``'none'``, then :math:`(N, *)`, same
+        - Input: :math:`(*)`. where :math:`*` means any number of dimensions.
```
Comment (jbschlosser) [inline torch/nn/modules/loss.py:686]: typo: comma instead of period before "where"

**Review by jbschlosser — APPROVED** [review]
LGTM!
